extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -105,45 +109,213 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     }
     this.storeService_.subscribe(
       StateProperty.PAGE_ATTACHMENT_STATE,
-      (isOpen) => this.onPageAttachmentStateUpdate_(isOpen)
+      (isOpen) => {
+        const shoppingData = this.storeService_.get(
```

Comment:
I know you're checking `if (isOnActivePage)` in all the methods, but I wouldn't mind either moving the check here, or duplicating the check if you want to be super safe (or have other code paths calling these methods)

Reply:
+1 to this